packages/react/src/primitives/TextField/__tests__/TextField.test.tsx

```diff
@@ -86,11 +86,19 @@ describe('TextField component', () => {
     });
 
     it('should set size and variation data attributes', async () => {
-      render(<TextField label="Field" size="small" variation="quiet" />);
+      render(
+        <TextField
+          label="Field"
+          size="small"
+          testId="testField"
+          variation="quiet"
+        />
+      );
 
-      const field = await screen.findByRole('textbox');
-      expect(field.dataset['size']).toBe('small');
-      expect(field.dataset['variation']).toBe('quiet');
+      const textField = await screen.findByTestId('testField');
+      const input = await screen.findByRole('textbox');
+      expect(textField).toHaveAttribute('data-size', 'small');
```

Comment:
I found a bug earlier in `TextField` that we don't pass `size` prop to the high-level `field`. We should test the sizing of the `field` as a whole, which includes label.